src/idl/include/idl/processor.h

```diff
@@ -116,6 +116,7 @@ struct idl_pstate {
       IDL_PARSE_UNKNOWN_ANNOTATION_APPL_PARAMS
     } state;
     void *yypstate; /**< state of Bison generated parser */
+    idl_retcode_t result; /**< return value from idl_parse */
```

Comment:
I purposely left out the result from the parser state. Initial versions by Frans did that too and that'd mean we have to make sure it's always correct because it's not immediately obvious when this one can be used. Scanner, parser, etc. I think I prefer adding a thread-local variable in `parser.y` and calling an additional function to invoke the parser that returns the value of said variable. Or perhaps extending the Bison parser signature to accept a pointer to a result value. It's somewhat slower and not as simple/elegant as simply returning a negative value, but overal limits the scope in which errors can be made in deciding when to use the return value and/or `result` member in the parser state (if that makes sense).